### PR TITLE
feat: add basic Desireset wrapper

### PR DESCRIPTION
**Pull request by JensSteenmetz**
## Description

I created a simple DesireSet with one single main GoalStructure. It is thus simply a wrapper for GoalStructure right now, as we do not need the complex DesireSet functionality for getting aplib to work.

## Testability

1. Check if you see errors in the functionality of the DesireSet class.
2. Check if documentation, identifier naming, and coding style are up to par.
3. Review the appropriateness of the unit tests.

## Checklist
- [x] PR #25 closed
- [x] Set the proper pull request name
- [x] Merged main into your branch

### CODE REVIEW

**Review by joachimdekker — CHANGES_REQUESTED** [review]
Discussed on Discord: extract UpdateStatus and put in the implementation of GetCurrentGoal.

**Review by joachimdekker — DISMISSED** [review]
After careful consideration, the previous requested changes were not desired. Hence, looks good to me.

File: Aplib.Core/Desire/DesireSet.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        /// Initializes a new instance of the <see cref="DesireSet{TBeliefSet}" /> class.
+        /// </summary>
+        /// <param name="mainGoal">The main goal structure that the agent needs to complete.</param>
+        public DesireSet(IGoalStructure<TBeliefSet> mainGoal) => _mainGoal = mainGoal;
+
+        /// <inheritdoc />
+        public IGoal GetCurrentGoal(TBeliefSet beliefSet)
+        {
+            return _mainGoal.GetCurrentGoal(beliefSet);
+        }
```
Comment (QuakeEye) [inline Aplib.Core/Desire/DesireSet.cs]:
```suggestion
        public IGoal GetCurrentGoal(TBeliefSet beliefSet)
            => _mainGoal.GetCurrentGoal(beliefSet);
```

**Review by QuakeEye — CHANGES_REQUESTED** [review]

File: Aplib.Core/Desire/DesireSet.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        /// <inheritdoc />
+        public IGoal GetCurrentGoal(TBeliefSet beliefSet)
+            => _mainGoal.GetCurrentGoal(beliefSet);
+
+        /// <inheritdoc />
+        public void UpdateStatus(TBeliefSet beliefSet)
+        {
+            _mainGoal.UpdateStatus(beliefSet);
+        }
```
Comment (joachimdekker) [inline Aplib.Core/Desire/DesireSet.cs]: Zou je hier ook een expression body van maken?


```suggestion
        public void UpdateStatus(TBeliefSet beliefSet)
            => _mainGoal.UpdateStatus(beliefSet);
```

**Review by joachimdekker — CHANGES_REQUESTED** [review]
Looks good, 1 minor style change.

**Review by QuakeEye — DISMISSED** [review]

**Review by QuakeEye — APPROVED** [review]

**Review by joachimdekker — APPROVED** [review]
LGTM

**Pull request merged**